apps/crawl-website/app/crawl_website.py

```diff
@@ -0,0 +1,347 @@
+        """Process the response from the request.
+
+        The resulting item contains a properties dictionary and a blob.
+        """
+        properties = {}
+
+        properties["url"] = response.url
+        logging.info(f"ApertureDBPipeline: Processing {response.url}")
+
+        content_type = response.headers.get(
+            "Content-Type", b"").decode("utf-8")
+        if content_type:
+            properties["content_type"] = content_type
+            simple_content_type = content_type.split(";")[0]
+            properties["simple_content_type"] = simple_content_type
+
+        last_modified = response.headers.get(
+            "Last-Modified", b"").decode("utf-8")
+        if last_modified:
+            last_modified_dt = email.utils.parsedate_to_datetime(last_modified)
+            properties["last_modified"] = {
+                "_date": last_modified_dt.isoformat()}
+
+        etag = response.headers.get("ETag", b"").decode("utf-8")
+        if etag:
+            properties["etag"] = etag
+
+        cache_control = response.headers.get(
+            "Cache-Control", b"").decode("utf-8")
+        if cache_control:
+            properties["cache_control"] = cache_control
+            # extract max-age from cache-control
+            max_age = None
+            for directive in cache_control.split(","):
+                if directive.strip().startswith("max-age"):
+                    max_age = int(directive.split("=")[1])
+                    break
+            if max_age:
+                properties["cache_control_max_age"] = max_age
+
+        expires = response.headers.get("Expires", b"").decode("utf-8")
+        if expires:
+            expires_dt = email.utils.parsedate_to_datetime(expires)
+            properties["expires"] = {
+                "_date": expires_dt.isoformat()}
+
+        # Not exactly right, but close enough for now
+        properties['crawl_time'] = {
+            "_date": datetime.now(timezone.utc).isoformat()}
+
+        return ApertureDBItem(properties=properties, blob=response.body)
+
+
+class ApertureDBPipeline:
+    """Pipeline to store items in ApertureDB"""
+
+    def __init__(self, db, crawl_id):
+        self.db = db
+        self.crawl_id = crawl_id
+        logging.info(f"ApertureDBPipeline: Using Crawl ID: {crawl_id}")
+
+    @classmethod
+    def from_crawler(class_, crawler):
+        db = CommonLibrary.create_connector()
+        args = crawler.settings.get("APERTUREDB_PIPELINE_ARGS", {})
+        crawl_id = crawler.settings.get("APERTUREDB_CRAWL_ID")
+
+        return class_(db, crawl_id)
+
+    def process_item(self, item: ApertureDBItem, spider):
+        query = [
+            {
+                "FindEntity": {
+                    "with_class": "Crawl",
+                    "constraints": {
+                        "id": ["==", self.crawl_id],
+                    },
+                    "_ref": 1,
+                }
+            },
+            {
+                "AddEntity": {
```

Comment:
Would there be any other use for this document?I was thinking moving the properties to the blob would also server the purpose. The sort of hierarchy would be useful later when we dabble into segmented text pieces.